py3status/modules/hueshift.py

```diff
@@ -0,0 +1,191 @@
+
+Requires:
+    blueshift: an extensible and highly configurable alternative
+    redshift: program to adjust the color temperature of your screen
+    sct: set color temperature
+
+Suggestions:
+    campfire: 4500
+    dust storm on mars: 2000
+    coffee free all nighter: 8000
+
+Examples:
+```
+# good rules; show command name; don't colorize everything
+hueshift {
+    format = '[\?if=disabled&color=bad \u263c!|[\?color=hue&show \u263c '
+    format += '{command}] {temperature}K]'
+    minimum = 3000
+    maximum = 10000
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+# warm
+{'full_text': u'\u263c 3000K', 'color': #f3c},
+
+# neutral
+{'full_text': u'\u263c 6500K', 'color': #ff3}
+
+# cool
+{'full_text': u'\u263c 10000K', 'color': #3cf},
+
+disabled
+{'full_text': u'\u263c!', 'color': #f00}
+"""
+
+STRING_BAD_COMMAND = 'invalid command `%s`'
+STRING_NOT_INSTALLED = 'command `%s` not installed'
+STRING_NOT_AVAILABLE = 'no available command'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_down = 5
+    button_reset = 3
+    button_up = 4
+    command = None
+    delta = 250
+    format = u'\?if=disabled&color=bad \u263c!|\?color=hue \u263c {temperature}K'
+    initial = 6500
+    maximum = 25000
+    minimum = 1000
+    thresholds = [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]
+
+    def post_config_hook(self):
+        hueshift_commands = ['blueshift', 'redshift', 'sct']
+        self.pgrep_command = ['pgrep', '|'.join(hueshift_commands)]
+        if not self.command:
+            self.command = self.py3.check_commands(hueshift_commands)
+        elif self.command not in hueshift_commands:
+            raise Exception(STRING_BAD_COMMAND % self.command)
+        elif not self.py3.check_commands(self.command):
+            raise Exception(STRING_NOT_INSTALLED % self.command)
+        if not self.command:
+            raise Exception(STRING_NOT_AVAILABLE)
+
+        self.hue = {
+            'blueshift': {
+                'set': lambda value: ['blueshift', '-p', '-t', str(value)],
+                'reset': lambda value: ['blueshift', '-p', '-r'],
+            },
+            'redshift': {
+                'set': lambda value: ['redshift', '-r', '-O', str(value)],
+                'reset': lambda value: ['redshift', '-r', '-x'],
+            },
+            'sct': {
+                'set': lambda value: ['sct', str(value)],
+                'reset': lambda value: ['sct']
+            },
+        }
+
+        self.minimum = max(self.minimum, 1000)
+        if self.maximum > 25000:
+            self.maximum = 25000
+        if self.command == 'sct' and self.maximum > 10000:
+            self.maximum = 10000
+        if not(self.minimum <= self.initial <= self.maximum):
+            self.initial = 6500
+
+        self.temperature = self.py3.storage_get(
+            'temperature') or self.initial
+        self.last_temperature = self.temperature
```

Comment:
this last_temperature has me confused,  what exactly are we using it for - because I think it is being missused